forte/processors/stave/stave_processor.py

```diff
@@ -0,0 +1,316 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+A StaveProcessor is introduced to enable immediate visualization of forte
+pipeline result. It supports serving Stave instance with StaveViewer.
+Forte users can plug it into the pipeline to easily visualize datapacks
+with annotations. It is also highly configurable for users to change port
+number, host name, layout, etc.
+
+Package Requirements:
+    forte
+    *stave (Required in future version)
+
+Required environment variables:
+*All of the variables below will be deprecated in future version.
+    FRONTEND_BUILD_PATH:
+        Absolute path (or relative path from PYTHONPATH)
+        to stave build folder. Example: "stave/build/"
+    PYTHONPATH:
+        Absolute path to django backend folder (e.g.,
+        "$STAVE_PATH/simple-backend/") should be inserted into PYTHONPATH.
+"""
+
+import os
+import json
+import logging
+import collections
+from typing import Dict, Set, Any
+import requests
+
+from forte.common import Resources, ProcessorConfigError
+from forte.common.configuration import Config
+from forte.data.data_pack import DataPack
+from forte.data.ontology.code_generation_objects import search
+from forte.processors.base import PackProcessor
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    "StaveProcessor"
+]
+
+
+class StaveProcessor(PackProcessor):
+    """
+    `StaveProcessor` provides easy visualization for forte users. We can
+    visualize datapack with annotations by inserting it into the forte
+    pipeline without affecting the original functionalities.
+
+    `StaveProcessor` requires an ontology file being passed to the pipeline.
+    It then genertes default configurations based on the input ontology to
+    start a stave instance without any additional specification by users.
+    Example usage:
+        pipeline.add(StaveProcessor())
+
+    `StaveProcessor` is also highly customizable for users to set up. Users may
+    configure port number, server host, project name, etc.
+    Example usage:
+        pipeline.add(StaveProcessor(), configs={
+            "port": 8880,
+            "project_name": "serialization_pipeline_test"
+        })
+
+    Users can modify project configs by changing the `project_configs` field.
+    Example usage:
+        pipeline.add(StaveProcessor(), configs={
+            "port": 8879,
+            "project_configs": {
+                # Configure Stave layout
+                "layoutConfigs": {
+                    "center-middle": "DialogueBox"
+                }
+            }
+        })
+    """
+
+    def __init__(self):
+        super().__init__()
+        self._project_id: int = -1
+        # TODO: Specify annotations in future update.
+        self._viewer: Any
+        self._project_writer: Any
```

Comment:
We should know the class for `project_writer` now.

Reply:
This will be updated after [the import statements](https://github.com/mylibrar/forte/blob/7719934e219f5b96887e082cb6bb89534960c206/forte/processors/stave/stave_processor.py#L119) are moved to the top-level. Otherwise, the annotations may no be identified here.